model.go

```diff
@@ -203,6 +203,17 @@ func (n *Node) Path() (out string) {
 	return strings.TrimSpace(out)
 }
 
+// ClosestGroup finds the first non-empty group in this node and its ancestors.
+func (n *Node) ClosestGroup() string {
```

Comment:
I initially thought that subcommands would automatically inherit the group of their parents, from there:

https://github.com/alecthomas/kong/blob/2479d83cc0307560579a16089433cbed7a6e5c77/build.go#L76-L79

But subcommands are actually escaping the `for` loop from here:

https://github.com/alecthomas/kong/blob/2479d83cc0307560579a16089433cbed7a6e5c77/build.go#L62

I didn't want to mess too much with this part, so I added this `ClosestGroup()` function to do the job.